src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CustomerSegmentHandler.php

```diff
@@ -0,0 +1,224 @@
+                            'applied_price' => $appliedPrice,
+                            'quantity' => $quantity,
+                        ]
+                    );
+
+                    $admin_email = get_option('admin_email');
+                    $subject = 'Price Mismatch Detected';
+                    $message = sprintf(
+                        'A mismatch was detected between the segment price and the applied price at checkout for product: %s (ID: %d). Segment Price: €%s vs Applied Price: €%s. Customer Email: %s, Quantity: %d',
+                        $product->get_name(),
+                        $product->get_id(),
+                        $segmentPrice,
+                        $appliedPrice,
+                        $order->get_billing_email(),
+                        $quantity
+                    );
+                    wp_mail($admin_email, $subject, $message);
+                }
+            }
+        }
+    }
+
+    /**
+     * @return float|mixed
+     */
+    public function getStandardPrice($product)
+    {
+        if ($product->is_type('variable')) {
+            $defaultVariation = $product->get_available_variations()[0];
+
+            return $defaultVariation['display_price'];
+        }
+
+        return (float) $product->get_regular_price();
+    }
+
+    /**
+     * @throws WordpressException
+     */
+    public function checkPriceMismatchOnCheckout($orderId): void
+    {
+        self::checkPriceMismatchOnOrder($orderId);
+        self::saveCustomerSegmentAsOrderMeta($orderId);
+    }
+
+    public function saveCustomerSegmentAsOrderMeta($orderId): void
+    {
+        $order = wc_get_order($orderId);
+        $findSegment = CustomerSegmentModel::findByUserId(get_current_user_id());
+
+        foreach ($order->get_items() as $itemId => $item) {
+            $productId = $item->get_product_id();
+            $qty = $item->get_quantity();
+            $findSegmentPrice = CustomerSegmentPriceModel::findByCustomerSegmentId($productId, $findSegment->customer_segment_id, $qty);
+            $segment = $findSegmentPrice;
+
+            if ($segment) {
+                update_post_meta($orderId, '_customer_segment_id', $segment->id);
+                update_post_meta($orderId, '_customer_segment_name', $segment->name);
+            } else {
+                update_post_meta($orderId, '_customer_segment_id', 'No segment found');
+                update_post_meta($orderId, '_customer_segment_name', 'No segment found');
+            }
+        }
+    }
+
+    public function showProductQtyBasedCart(): void
+    {
+        global $woocommerce;
+
+        if (is_product()) {
+            $productId = get_the_ID();
+        } elseif (is_cart()) {
+            $productId = null;
+        }
+
+        $quantity = 0;
+
+        if (is_cart()) {
+            foreach ($woocommerce->cart->get_cart() as $cartItemKey => $cartItem) {
+                if (null === $productId || $cartItem['product_id'] === $productId) {
```

Comment:
why do we sum all different products on cart as well with `null === $productId` ?